### DISCUSSION TITLE
Generic "Shopify Payments" name creates issues for customizing payment options

### DISCUSSION THREAD

**Discussion by kylebuildsstuff** [Shopify/function-examples#338] [Feedback: Payment Customization API]
Reordering payment methods works for most cases as I can just sort based on name, but I just realized that for payment methods like "Credit card" or non-manual type payment options the name is always "Shopify Payments". This makes it extremely difficult, maybe impossible? to reliably customize these payment options.

For example, here's the input I get:

```
paymentMethods": [
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/0",
      "name": "Shopify Payments"
    },
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/1",
      "name": "Shopify Payments"
    },
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/2",
      "name": "Shopify Payments"
    },
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/3",
      "name": "Cash on Delivery (COD)"
    },
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/4",
      "name": "Money Order"
    },
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/5",
      "name": "Bank Deposit"
    },
    {
      "id": "gid://shopify/PaymentCustomizationPaymentMethod/6",
      "name": "Pay at store"
    }
  ],
```

Instead of 3x "Shopify Payments" can we have more specific names, or unique identifiers we can hook into? Or is there some documentation or rules surrounding the ordering of these "Shopify Payments" payment methods, like gid://shopify/PaymentCustomizationPaymentMethod/0 refers to "Credit card"?

**Comment by maks-afew** [comment]
Bump

**Comment by maks-afew** [comment]
Bump

**Reply by davejcameron** [reply]
@maks-afew have you tried the latest version of the Functions API? It should address these issues.

**Reply by maks-afew** [reply]
@davejcameron it seems to work now, nice 👍
